### ISSUE TITLE
Change of container UID/GID for shiny user (away from 999:999)

### ISSUE THREAD

**Issue by ds-04** [rocker-org/shiny#88]
Currently the UID/GID of the shiny user within the container appears to be 999 which is a bit problematic on Centos/RHEL systems.

I'm currently trying to use the container on Centos 8.2.

Would it be possible to change the UID/GID to something else? (I'm finding in Centos 8.2 various things are occupying 990-999.

I've tried adapting the docker file to do this, but I can't get it to build at the moment (using podman 1.6.4).

It would be great if this change was in the container.

**Comment by ds-04** [comment]
Something along these lines, after the package installs:<br>
`RUN groupadd --gid 5001 shiny && useradd --home-dir /home/shiny --create-home --uid 5001 --gid 5001 --shell /bin/sh --skel /dev/null`
<br><br>
perhaps also, before CMD:<br>
`USER shiny`

**Comment by cboettig** [comment]
thanks @ds-04 .  I'm worried about hardwiring a different uuid for users binding volumes, since the change can break permissions.  but maybe at least some of this can pre-configured to make it easier to opt into.  
@noamross thoughts on this?

**Comment by ds-04** [comment]
Yes that makes sense. Perhaps two docker files can be provided. One with hardwired uids/gids, one without.

Just edited the above comment as I'd missed off _shiny_ at the end of the useradd line.

Managed to get it built with those changes, though I'm still exploring docker and podman for this.

thanks

**Comment by cboettig** [comment]
thanks.  yeah, one day I'd love to add some more explicit `podman` instructions to our docs over https://rstudio-project.org (or maybe the wiki https://github.com/rocker-org/rocker/wiki; our dev team doesn't use podman right now but it comes up frequently in issues and it would be good to distil the collective community wisdom!  If you do write up any notes on your podman exploration please feel free to share.  We're very much driven by community contributions here.

**Comment by ds-04** [comment]
Seems to be reasonably happy running on Centos8 podman for awhile now.

Some notes.

podman version 2.2.1
CentOS Linux release 8.3.2011

Some excerpts from Docker file used by podman on our VM:

`RUN groupadd --gid 5001 shiny && useradd --home-dir /home/shiny --create-home --uid 5001 --gid 5001 --shell /bin/sh --skel /dev/null shiny`

before CMD at the end of the file:

`USER shiny`


-------------



Building

`podman build --network=host -t <ImageName> `

Running

`podman run --name <ContainerName> --detach -p 3838:3838 <ImageName>`
